### PR TITLE
feat: Current location feature in welcome screen

### LINKED ISSUE #911: Add current location option in welcome screen.
**Describe the feature you'd like**

Add option for the current location in the welcome screen.

**Screenshots**

current

**Additional context**

@nikit19 @iamareebjamal @liveHarshit Your views?

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by kshitijsuri90**
Fixes #911 #942

Changes: 
Added option for current location with revamps in the icon.

Screenshots for the change:

Is this correct @iamareebjamal @nikit19 @liveHarshit?

### CODE REVIEW

**Comment by liveHarshit** [comment]
I don't think we need to copy exactly. Current location is available with the location fragment. No need to configure it again as I already told in the issue comment.
@iamareebjamal @nikit19 Your views?

File: app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt [hunk 1]
```diff
@@ -23,6 +37,53 @@ class WelcomeFragment : Fragment() {
...
+        val permission =
+            ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION)
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST)
+        }
+    }
+
+    private fun redirectToMain() {
+        val startMainActivity = Intent(activity, MainActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt]: It does not work in the same way as should. Activity is preparing again with slash theme.

**Comment by kshitijsuri90** [comment]
Yes, an option is provided in pick a city too but its ease of accessibility to the user is provided in the initial welcome fragment. @liveHarshit.

File: app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt [hunk 2]
```diff
@@ -21,6 +37,51 @@ class WelcomeFragment : Fragment() {
...
+            ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION)
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST)
+        }
+    }
+
+    private fun redirectToMain() {
+        val startMainActivity = Intent(activity, MainActivity::class.java)
+        activity?.startActivity(startMainActivity)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt]: Same. Why don't you use navigation controller?

Comment (kshitijsuri90) [reply app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt]: Okay will do that.

Comment (kshitijsuri90) [reply app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt]: @liveHarshit On doing that I am losing the action bar as not provided in the welcome fragment. Any suggestions how to solve that?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt]: Sorry, cannot get it?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt]: Add `thisActivity.supportActionBar?.show()` inside `if (thisActivity is AppCompatActivity)` in onCreateView of events fragment.

**Comment by liveHarshit** [comment]
Update GIF.

**Comment by kshitijsuri90** [comment]
Done @liveHarshit @nikit19. Please Review.

**Comment by kshitijsuri90** [comment]
@iamareebjamal @nikit19 Please do review.

**Pull request merged**